library/core/src/num/f64.rs

```diff
@@ -1389,7 +1389,10 @@ impl f64 {
     #[stable(feature = "clamp", since = "1.50.0")]
     #[inline]
     pub fn clamp(mut self, min: f64, max: f64) -> f64 {
-        assert!(min <= max);
+        assert!(
+            min <= max,
+            "min > max, or either was NaN. This is not allowed. min = {min:?}, max = {max:?}"
+        );
```

Comment:
These messages may be seen by the end user, but they're typically a programming error, so let's avoid phrasing that might be misinterpreted. An assertion failure already makes it clear something wasn't allowed.
```suggestion
        assert!(
            min <= max,
            "min > max, or either was NaN. min = {min:?}, max = {max:?}"
        );
```

Reply:
Done